cogent/lib/cogent-endianness.h

```diff
@@ -0,0 +1,54 @@
+#ifndef __COGENT_ENDIANNESS_H__
+#define __COGENT_ENDIANNESS_H__
+
+const int ENDIANNESS_TEST = 1;
+#define IS_BIG_ENDIAN (*(char *)&ENDIANNESS_TEST == 0)
+
+u16 swap_u16(u16 v) {
+  return (v << 8) | (v >> 8);
+}
+
+u32 swap_u32(u32 v) {
```

Comment:
GNU provides swapping functions (`bswap_*`) in the <byteswap.h> header file. Might it be a good idea to use these functions instead?

Reply:
@cmcl I'm not sure about it. This code will be read by C-parser/AutoCorres; libc headers usually use extensions/features that we can't read.